src/BackendPkg/Database.go

```diff
@@ -446,20 +447,48 @@ func (d *Database) WriteNewUserRecipe (currUser User, newRecipe Recipe){
 	statement, _ := database.Prepare("CREATE TABLE IF NOT EXISTS UserRecipeData (title TEXT, ingredients TEXT, instructions TEXT, recipeID TEXT PRIMARY KEY, username TEXT)")
 	statement.Exec()
 
-	// find num of recipes made by user to generaate userID
-	idNum := 0
-	err := database.QueryRow("SELECT COUNT(*) FROM UserRecipeData WHERE username = ?", currUser.UserName).Scan(&idNum)
-	if err != nil {idNum = 1 } // if user has no recipes then idNum = 1
-
 	// Insert each recipe into the table
 	ingredients, _ := json.Marshal(newRecipe.Ingredients)
 	statementThree, _ := database.Prepare("INSERT OR IGNORE INTO UserRecipeData (title, ingredients, instructions, recipeID, username) values (?, ?, ?, ?, ?)")
-	statementThree.Exec(newRecipe.Title, string(ingredients), newRecipe.Instructions, (currUser.UserName + strconv.Itoa(idNum)), currUser.UserName)
+	statementThree.Exec(newRecipe.Title, string(ingredients), newRecipe.Instructions, (currUser.UserName + strconv.Itoa(d.getNextRecipeID(currUser.UserName))), currUser.UserName)
```

Comment:
So all the recipes go off the same id?

Reply:
no, it's <username> + <highest number recipe + 1>. if you make 5 recipes as a new user it would be like ri1, ri2, ri3, ri4, and ri5, but if I have ri2, ri300, and ri60 in the database (after deletes), the next one i make will be ri301.